src/draw/lv_image_decoder.c

```diff
@@ -151,6 +151,12 @@ lv_result_t lv_image_decoder_open(lv_image_decoder_dsc_t * dsc, const void * src
      * */
     lv_result_t res = dsc->decoder->open_cb(dsc->decoder, dsc);
 
+    /*Free the source if it's a file*/
+    if(dsc->src_type == LV_IMAGE_SRC_FILE) {
+        lv_free((void *)dsc->src);
+        dsc->src = NULL;
```

Comment:
If it's never used later, then let's remove it from dsc, and need to dup it at beginning.

Reply:
You're right